packages/metrics-discovery/src/metrics-discovery.ts

```diff
@@ -0,0 +1,53 @@
+import type { NodeStructOutput } from '@river-build/generated/dev/typings/INodeRegistry'
+import { RiverRegistry, getWeb3Deployment } from '@river-build/web3'
+import { ethers } from 'ethers'
+
+const NODE_STATUS = {
+    ACTIVE: 2,
+    DEPARTING: 4,
+}
+
+// This class is in charge of pulling all nodes from river and
+// formatting them into a prometheus-friendly format
+
+export class MetricsDiscovery {
+    constructor(private readonly riverRegistry: RiverRegistry, private readonly env: string) {}
+
+    public static init(config: { riverRpcURL: string; env: string }) {
+        const deployment = getWeb3Deployment(config.env)
+        const provider = new ethers.providers.JsonRpcProvider(config.riverRpcURL)
+        const riverRegistry = new RiverRegistry(deployment.river, provider)
+        return new MetricsDiscovery(riverRegistry, config.env)
+    }
+
+    public static isTargeted(node: NodeStructOutput) {
+        return node.status === NODE_STATUS.ACTIVE || node.status === NODE_STATUS.DEPARTING
```

Comment:
I think we should include Failed nodes in the set given stream replication is not yet implemented so the network failure case exists whereby an operational node -> failed and the streams it hosts are broken.  I understand that a Failed node may not be running to emit metrics or may be running in a partially failed state or may be running correctly and be mislabelled by the operator as Failed. Doesn't hurt to attempt scraping from these nodes in my view.

Reply:
good feedback 👍 adding failed nodes